Emrald_Site/scripts/UI/Sidebar.js

```diff
@@ -1749,6 +1736,12 @@ if (typeof Navigation === 'undefined')
       var contentPanel = document.getElementById("ContentPanel");
       adjustWindowPos(contentPanel, wnd.div);
       contentPanel.appendChild(wnd.div);
+      }
+    //-------------------------------------------
+    Sidebar.prototype.editDiagramType = function (diagramName, oldType, newType) {
+      let successful = false;
+      console.log('Changing Diagram Type');
+      return successful;
     }
```

Comment:
@pressr-inl Here is where you can implement the Diagram Type changing logic. This function should return a boolean. 
- If it returns `true`, then it will allow the user to change the Type
- If it returns `false`, then it will not allow the user to change the Type and display an alert saying so